tests/csm/rest/test_iam_users.py

```diff
@@ -2126,3 +2126,207 @@ def test_39217(self):
+        self.log.info("Create bucket successful for user")
+        test_file = "test-object.txt"
+        file_path_upload = os.path.join(TEST_DATA_FOLDER, test_file)
+        if os.path.exists(file_path_upload):
+            os.remove(file_path_upload)
+        if not os.path.isdir(TEST_DATA_FOLDER):
+            self.log.debug("File path not exists, create a directory")
+            system_utils.execute_cmd(cmd=common_cmd.CMD_MKDIR.format(TEST_DATA_FOLDER))
+        system_utils.create_file(file_path_upload, self.file_size)
+        self.log.info("Step: Verify put object.")
+        resp = s3_obj.put_object(bucket_name=bucket_name, object_name=test_file,
+                                 file_path=file_path_upload)
+        self.log.info("Removing uploaded object from a local path.")
+        os.remove(file_path_upload)
+        assert_utils.assert_true(resp[0], resp[1])
+        self.log.info("Step: Verify get object.")
+        resp = s3_obj.get_object(bucket_name, test_file)
+        assert_utils.assert_true(resp[0], resp)
+        self.log.info("Step 6: Delete bucket")
+        resp = s3_obj.delete_bucket(bucket_name=bucket_name, force=True)
+        assert_utils.assert_true(resp[0], resp[1])
+        self.log.info("Bucket deleted successfully")
+        self.log.info("##### Test ended -  %s #####", test_case_name)
+
+    @pytest.mark.lc
+    @pytest.mark.csmrest
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.tags('TEST-39028')
+    def test_39028(self):
+        """
+        Add-remove invalid capabilities
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info(
+            "Step 1: Login using csm user and create a user with some capabilities")
+        payload = self.csm_obj.iam_user_payload_rgw(user_type="valid")
+        user_cap = "usage=read;buckets=write;users=read"
+        payload.update({"user_caps": user_cap})
+        self.log.info("payload :  %s", payload)
+        resp1 = self.csm_obj.create_iam_user_rgw(payload)
+        assert resp1.status_code == HTTPStatus.CREATED, \
+            "User could not be created"
+        uid = resp1.json()['tenant'] + "$" + payload["uid"]
+        self.log.info("Step 2: Add some invalid capabilities")
+        user_cap = "random=;buckets="
+        payload = {}
+        payload.update({"user_caps": user_cap})
+        resp = self.csm_obj.add_user_caps_rgw(uid, payload)
+        self.log.info("Verify Response : %s", resp)
+        assert_utils.assert_true(resp.status_code == HTTPStatus.BAD_REQUEST,
+                                     "Invalid caps added")
+        get_resp = self.csm_obj.get_iam_user(user=uid)
+        assert_utils.assert_true(get_resp.status_code == HTTPStatus.OK, "Get IAM user failed")
+        self.log.info("STEP 3: Verify added capabilities")
+        diff_items = self.csm_obj.verify_caps(user_cap, get_resp.json()["caps"])
```

Comment:
here user_cap should be initial caps, but user_cap is getting updated on line 2307.

Reply:
changed verification code